slsReceiverSoftware/src/MasterAttributes.h

```diff
@@ -0,0 +1,567 @@
+#pragma once
+
+#include "ToString.h"
+#include "logger.h"
+#include "sls_detector_defs.h"
+
+#ifdef HDF5C
+#include "H5Cpp.h"
+#ifndef H5_NO_NAMESPACE
+using namespace H5;
+#endif
+#endif
+
+#include <chrono>
+using ns = std::chrono::nanoseconds;
+
+// versions
+#define HDF5_WRITER_VERSION   (6.1) // 1 decimal places
+#define BINARY_WRITER_VERSION (6.1) // 1 decimal places
+
+class MasterAttributes {
+  public:
+    slsDetectorDefs::detectorType detType{slsDetectorDefs::GENERIC};
+    slsDetectorDefs::timingMode timingMode{slsDetectorDefs::AUTO_TIMING};
+    uint32_t imageSize{0};
```

Comment:
Do we want all the data fields to be declared in the base class or only the ones that are common for all detectors?

Reply:
They cannot be called by the parent reference that we are using